### PR TITLE
Use autocfg to determine rust version

### LINKED ISSUE #2229: Use version_check for minimum rust version
While doing #2228 I noticed that we have a dependency on [version_check](https://crates.io/crates/version_check). We might want to use this for the minimum rust version in `xtask`.

### PR DESCRIPTION
**Pull request by kjeremy**
Fixes #2229

### CODE REVIEW

File: xtask/src/main.rs [hunk 1]
```diff
@@ -226,20 +227,14 @@ fn install_client(ClientOpt::VsCode: ClientOpt) -> Result<()> {
...
-    let mut old_rust = false;
-    if let Ok(output) = run_with_output("cargo --version", ".") {
-        if let Ok(stdout) = String::from_utf8(output.stdout) {
-            println!("{}", stdout);
-            if !check_version(&stdout, REQUIRED_RUST_VERSION) {
-                old_rust = true;
-            }
-        }
-    }
+    let ac = autocfg::AutoCfg::with_dir("target")?;
```
Comment (kjeremy) [inline xtask/src/main.rs]: I think this is right cc: @matklad

File: xtask/src/main.rs [hunk 2]
```diff
@@ -9,17 +9,18 @@
...
 use std::{env, path::PathBuf};
 use xtask::{
     codegen::{self, Mode},
-    install_format_hook, run, run_clippy, run_fuzzer, run_rustfmt, run_with_output, Cmd, Result,
+    install_format_hook, run, run_clippy, run_fuzzer, run_rustfmt, Cmd, Result,
 };
 
 // Latest stable, feel free to send a PR if this lags behind.
-const REQUIRED_RUST_VERSION: u32 = 39;
+const REQUIRED_RUST_VERSION: (usize, usize) = (1, 39);
```
Comment (matklad) [inline xtask/src/main.rs]: Horrible idea `: f32 = 1.39` (No, we won’t do this)

File: xtask/src/main.rs (hunk 1 above)
Comment (matklad) [reply xtask/src/main.rs]: Seems fine!

File: xtask/src/main.rs (hunk 2 above)
Comment (lnicola) [reply xtask/src/main.rs]: `1 * 1_000 + 39` :laughing:

**Pull request merged**